store/metrics.go

```diff
@@ -130,15 +138,27 @@ func (m *metrics) observeHas(ctx context.Context, dur time.Duration, failed bool
 		attribute.Bool(failedKey, failed)))
 }
 
-func (m *metrics) observeRemove(ctx context.Context, dur time.Duration, failed bool) {
+func (m *metrics) observeRemoveAll(ctx context.Context, dur time.Duration, failed bool) {
+	if m == nil {
+		return
+	}
+	if ctx.Err() != nil {
+		ctx = context.Background()
```

Comment:
So, always blocking? Maybe `context.WithTimeout(context.Background(), time.Second)` ?

Reply:
Good that you pointed out. We use common function to reset context in metrics everywhere else. I'll change how we do it in another PR.